NOTES.ANDROID

```diff
@@ -26,15 +26,22 @@
  you need to set ANDROID_NDK_HOME environment to point at NDK directory
  as /some/where/android-ndk-<ver>. Both variables are significant at both
  configuration and compilation times. NDK customarily supports multiple
- Android API levels, e.g. android-14, android-21, etc. By default latest
+ Android API levels, e.g. android-16, android-29, etc. By default latest
  one available is chosen. If you need to target older platform, pass
  additional -D__ANDROID_API__=N to Configure. N is numeric value of the
- target platform version. For example, to compile for ICS on ARM with
- NDK 10d:
+ target platform version. 
+ If you have Android Studio installed then the NDK is in 
+ $HOME/Android/Sdk/ndk-bundle.
 
-    export ANDROID_NDK_HOME=/some/where/android-ndk-10d
-    PATH=$ANDROID_NDK_HOME/toolchains/arm-linux-androideabi-4.8/prebuilt/linux-x86_64/bin:$PATH
-    ./Configure android-arm -D__ANDROID_API__=14
+ For example, to compile for target aarch64-linux-android using NDK r19c:
+
+    export ANDROID_NDK_HOME=$HOME/Android/Sdk/ndk-bundle
+    NDK=$ANDROID_NDK_HOME \
+    HOST_TAG=linux-x86_64 \
+    PATH=$NDK/toolchains/llvm/prebuilt/$HOST_TAG/bin:$PATH \
+    CC=$NDK/toolchains/llvm/prebuilt/$HOST_TAG/bin/aarch64-linux-android28-clang \
+    CXX=$NDK/toolchains/llvm/prebuilt/$HOST_TAG/bin/aarch64-linux-android28-clang++ \
```

Comment:
Why these two?  I thought clang selected its target architecture with something like `-target aarch64-linux-android28`?, which our android target code already handles...  I understand that you want clang, but in that case, shouldn't the following be enough?

```
    CC=clang CXX=clang++
```

Reply:
I think the target is without the API level `aarch64-linux-android` but the name of the clang compiler has now the 28 in it. If you use the ndk-bundle there is no aarch64-linux-android-clang but aarch64-linux-android16-clang to aarch64-linux-android29-clang.
If you use `make-standalone-toolchain.sh` the aarch64-linux-android-clang wrapper script(s) are created.
Regarding 'why not just set clang and clang++?': Configure complains if unprefixed tools are used, if I remember correctly. Also Configure falls back to gcc quite fast and so best to state what should be used.

Discussion about gcc deprecation is here: https://github.com/android-ndk/ndk/issues/26
gcc deprecation timeline: https://android.googlesource.com/platform/prebuilts/clang/host/linux-x86/+/master/GCC_4_9_DEPRECATION.md

Reply:
There's code in `Configurations/15-android.conf` that goes to quite some length to make it easier.  Maybe it needs adapting?

Reply:
This PR is for the NOTES only. Yes, I think it makes sense to change the Android conf as well.
I wanted to keep this small for now and do the other steps later.

Reply:
I have sent the CLA to openssl legal. What is the way forward?